app/lib/actions/katello/alternate_content_source/create.rb

```diff
@@ -7,7 +7,9 @@ class Create < Actions::EntryAction
         def plan(acs, smart_proxies, products = nil)
           acs.save!
           action_subject(acs)
-          acs.products << products if products.present?
+          if products.present?
+            acs.update!(products: products)
+          end
```

Comment:
If `acs` previously had products,
```rb
acs.products << products
```
would add `products` to its existing products.

But 
```rb
acs.update!(products: products)
```
won't preserve existing products. Is this intended?

Reply:
This is happening on create so I don't believe either approach would make a difference?

Reply:
ok, sounds good!

Does `update!` work with `products` as an attribute like that? Since it's an association and not a column on the alternate_content_sources table, I wonder if `products = ` isn't more correct. But I may just be nitpicking. If it works I'm fine with either.

Reply:
`update!` is used so we can validate the list of products. `acs.products << products` and `acs.products = products` wouldn't validate until another `acs.save!` was called.

Reply:
ahhh good point! Carry on then :)